aspnet/fundamentals/static-files.rst

```diff
@@ -47,132 +50,88 @@ Now, let's say that you have a project hierarchy where the static files you wish
-  public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory)
-  {
-    ...
-    // Add MyStaticFiles static files to the request pipeline.
-    app.UseStaticFiles(new StaticFileOptions()
-    {
-        FileProvider = new PhysicalFileProvider(@"D:\Source\WebApplication1\src\WebApplication1\MyStaticFiles"),
-        RequestPath = new PathString("/StaticFiles")
-    });
-    ...
+A request to ``http://<yourApp>/StaticFiles/test.png`` will serve the *test.png* file.
 
-At this point, if the user enters an address of ``http://<yourApp>/StaticFiles/test.png``, the ``test.png`` image will be served.
+Static file authorization
+-------------------------
+
+The static file module provides **no** authorization checks. Any files served by it, including those under *wwwroot* are publicly available. If you want to serve files based on authorization you should:
+
+- Store them outside of *wwwroot* and any directory accessible to the static file middleware
+- Serve them through a controller action, returning a :dn:class:`~Microsoft.AspNetCore.Mvc.FileResult` where authorization is applied
 
 Enabling directory browsing
 ---------------------------
 
-Directory browsing allows the user of your Web app to see a list of directories and files within a specified directory (including the root). By default, this functionality is not available such that if the user attempts to display a directory within an ASP.NET Web app, the browser displays an error. To enable directory browsing for your Web app, call the ``UseDirectoryBrowser`` extension method from  ``Startup.Configure`` as follows:
-
-.. code-block:: c#
-  :emphasize-lines: 5
+Directory browsing allows the user of your web app to see a list of directories and files within a specified directory. Directory browsing is disabled by default for security reasons. To enable directory browsing, call the ``UseDirectoryBrowser`` extension method from  ``Startup.Configure`` as follows:
 
-  public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory)
-  {
-    ...
-    // Turn on directory browsing for the current directory.
-    app.UseDirectoryBrowser();
-    ...
+.. literalinclude:: static-files/sample/Startup.cs
+  :language: c#
+  :lines: 38-39,53,54,64-69
+  :emphasize-lines: 5-11
+  :dedent: 8
 
-The following figure illustrates the results of browsing to the Web app's ``images`` folder with directory browsing turned on:
+The code above allows directory browsing of the *wwwroot/images* folder:
 
 .. image:: static-files/_static/dir-browse.png
 
-Now, let's say that you have a project hierarchy where you want the user to be able to browse a directory that is not in the webroot. For example, let's take a simple layout like the following:
-
-  - wwwroot
-
-    - css
-    - images
-    - ...
-
-  - MyStaticFiles
-
-In order for the user to browse the ``MyStaticFiles`` directory, you can configure the static files middleware as follows:
```

Comment:
Why was this sample removed? At the very least you can say that the location and url can be configured in the same way as UseStaticFiles.